### PR TITLE
Make directory.HandleOps() unexported method of keyserver/listener.go

### PR DESCRIPTION
**Pull request by masomel**
While writing the API documentation for the protocol package, I thought that HandleOps is really a server-level function. This change makes the directory API strictly about operations on the data structure.

### CODE REVIEW

**Comment by c633** [comment]
The main reason of having `HandleOps` is to avoid the developers mistakenly call the handlers with a bad request (think about what we did before #108). I'm still think that these validations should belong to the protocol.

**Comment by masomel** [comment]
> The main reason of having `HandleOps` is to avoid the developers mistakenly call the handlers with a bad request (think about what we did before #108)

We definitely want to avoid developers misusing the handlers, but a `HandleOps` function that determines which handler to call based on the request type is really a server-level function since the directory is just a high-level data structure. If we want to validate the requests, why can't each handler fail early if the request is malformed? Then we can call `handleOps` from the server, which decides which handler to call.

**Comment by c633** [comment]
Yeah, we can do that way. LGTM.

File: keyserver/listener.go [hunk 1]
```diff
@@ -86,6 +86,31 @@ func malformedClientMsg(err error) ([]byte, error) {
 	return res, err
 }
 
+// andleOps validates the request message and then pass it to
```
Comment (c633) [inline keyserver/listener.go]: s/andleOps/handleOps

**Comment by masomel** [comment]
> Can we make `handleOps` method of `ConiksServer`? I gave this a shot in 227a2ab.

Sure, this LGTM! We can cherry-pick 227a2ab into this branch, not sure if that would be the easiest way to incorporate your changes.

**Comment by c633** [comment]
Yes, you can cherry pick.

**Pull request closed without merging**